.github/workflows/R-CMD-check-build.yaml

```diff
@@ -0,0 +1,68 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
+
+on:
+  workflow_call:
+
+name: R-CMD-check
+
+permissions: read-all
+
+jobs:
+  R-CMD-check:
+    runs-on: ${{ matrix.config.os }}
+
+    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: windows-latest, r: 'release'}
+          - {os: ubuntu-latest,   r: 'release'}
+
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      R_KEEP_PKG_SOURCE: yes
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: r-lib/actions/setup-pandoc@v2
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: ${{ matrix.config.r }}
+          http-user-agent: ${{ matrix.config.http-user-agent }}
+          use-public-rspm: true
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: any::rcmdcheck
+          needs: check
+
+      - uses: r-lib/actions/check-r-package@v2
+        with:
+          upload-snapshots: true
+          build_args: 'c("--no-manual","--compact-vignettes=gs+qpdf")'
+          error-on: 'c("error")'
```

Comment:
I wonder if we should add `--no-vignettes` to the **check** args?
`args: c("--no-manual", "--as-cran", "--no-vignettes")`
Otherwise all vignettes outputs are created twice: first during the check step, and then during the build step.

Reply:
It depends if we want to make vignettes available for the users directly from R (with `vignette(...)`) or not.

Reply:
I don't quite understand this.
If the vignettes are built and delivered as part of the package: they can be accessed via `vignette(..)`.. or not?

Reply:
Yes, but if they are not build during this process, they will not be available.

Reply:
Anyway, I think we should stick with the default behavior (this is the default from r-lib repo)

Reply:
Ok, I have created a separate issue for this: https://github.com/Open-Systems-Pharmacology/Workflows/issues/34